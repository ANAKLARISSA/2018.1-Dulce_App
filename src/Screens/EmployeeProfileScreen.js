import React from 'react';
import {View , Text} from 'react-native';
import AGRButton from '../Components/AGRButton';

const styles = {

container: {
flex: 1,
flexDirection: 'column',
padding: 50,
backgroundColor:'#FFF'
},
text: {
fontSize: 36,
marginBottom: 15,
alignSelf: 'center'
},
informacoes: {
  alignSelf: 'center'
}

}

class EmployeeProfileScreen extends React.Component{

  constructor(props){
    super(props);
    this.state = {
      employee: [],
      loading: true

    }

  }

  componentDidMount(){

    const url = 'https://jsonplaceholder.typicode.com/users';
    this.setState({loading: true});
    fetch(url)
    .then((response) => {return response.json();})
    .then((json) => {this.setState({employee: json,loading: false});})
  }

  render(){
    return(
      <View style={styles.container}>
      {
        this.state.loading ? (<Text style={styles.text}>Carregando...</Text>) : (
          <View style={styles.informacoes}>
          <Text style={styles.text}>{this.state.employee[0].name}</Text>
          <Text style={styles.text}>CRM/Matricula:</Text>
          <Text style={styles.text}>{this.state.employee[0].address.zipcode}</Text>
          <Text style={styles.text}>setor: Pediatria</Text>
          <Text style={styles.text}>Cirurgi√£o</Text>
          <Text style={styles.text}>Hospital do Gama</Text>
          <View style={{height: 50}} />
          <AGRButton
          text='editar'
          onPress={() => this.props.navigation.navigate('edit')} />
          </View>


        )

      }
      </View>
    )

  }



}

export default EmployeeProfileScreen;
