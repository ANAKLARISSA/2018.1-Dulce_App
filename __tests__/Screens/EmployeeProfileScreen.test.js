import React from 'react';
import 'react-native';
import EmployeeProfileScreen from '../../src/Screens/EmployeeProfileScreen';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
require('bezier');

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer
  .create(<EmployeeProfileScreen />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

test('render data',() => {
  const wrapper = shallow(<EmployeeProfileScreen />);
  const employee_example = {
    name: 'Doctor Example',
    registration: '123456',
    sector: 'Sector Example'
  };
  wrapper.setState({employee: employee_example});
  wrapper.setState({loading: false});

  const nameField = wrapper.find('Text').at(0);
  const nameFieldText = nameField.dive().text();
  expect(nameFieldText).toBe(employee_example.name);

  const registrationField = wrapper.find('Text').at(1);
  const registrationFieldText = registrationField.dive().text();
  expect(registrationFieldText).toBe('CRM/Matricula:' + employee_example.registration);

  const sectorField = wrapper.find('Text').at(2);
  const sectorFieldText = sectorField.dive().text();
  expect(sectorFieldText).toBe('setor: ' + employee_example.sector);

});
