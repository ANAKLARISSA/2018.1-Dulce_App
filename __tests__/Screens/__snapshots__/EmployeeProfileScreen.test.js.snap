// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmployeeProfileScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <View
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 30,
              "marginBottom": 15,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 20,
              "marginBottom": 15,
            }
          }
        >
          CRM/Matricula:
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 20,
              "marginBottom": 15,
            }
          }
        >
          setor: 
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 20,
              "marginBottom": 15,
            }
          }
        >
          Cirurgi達o
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "fontSize": 20,
              "marginBottom": 15,
            }
          }
        >
          Hospital do Gama
        </Text>
        <View
          style={
            Object {
              "height": 50,
            }
          }
        />
        <AGRButton
          onPress={[Function]}
          text="editar"
        />
      </View>,
      "style": Object {
        "backgroundColor": "#FFF",
        "flex": 1,
        "flexDirection": "column",
        "padding": 50,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 30,
                "marginBottom": 15,
              }
            }
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              }
            }
          >
            CRM/Matricula:
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              }
            }
          >
            setor: 
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              }
            }
          >
            Cirurgi達o
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              }
            }
          >
            Hospital do Gama
          </Text>,
          <View
            style={
              Object {
                "height": 50,
              }
            }
          />,
          <AGRButton
            onPress={[Function]}
            text="editar"
          />,
        ],
        "style": Object {
          "alignSelf": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": undefined,
            "ellipsizeMode": "tail",
            "style": Object {
              "alignSelf": "center",
              "fontSize": 30,
              "marginBottom": 15,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              "CRM/Matricula:",
              undefined,
            ],
            "ellipsizeMode": "tail",
            "style": Object {
              "alignSelf": "center",
              "fontSize": 20,
              "marginBottom": 15,
            },
          },
          "ref": null,
          "rendered": Array [
            "CRM/Matricula:",
            undefined,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": Array [
              "setor: ",
              undefined,
            ],
            "ellipsizeMode": "tail",
            "style": Object {
              "alignSelf": "center",
              "fontSize": 20,
              "marginBottom": 15,
            },
          },
          "ref": null,
          "rendered": Array [
            "setor: ",
            undefined,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Cirurgi達o",
            "ellipsizeMode": "tail",
            "style": Object {
              "alignSelf": "center",
              "fontSize": 20,
              "marginBottom": 15,
            },
          },
          "ref": null,
          "rendered": "Cirurgi達o",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Hospital do Gama",
            "ellipsizeMode": "tail",
            "style": Object {
              "alignSelf": "center",
              "fontSize": 20,
              "marginBottom": 15,
            },
          },
          "ref": null,
          "rendered": "Hospital do Gama",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "height": 50,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
            "text": "editar",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <View
          style={
            Object {
              "alignSelf": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 30,
                "marginBottom": 15,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              }
            }
          >
            CRM/Matricula:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              }
            }
          >
            setor: 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              }
            }
          >
            Cirurgi達o
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              }
            }
          >
            Hospital do Gama
          </Text>
          <View
            style={
              Object {
                "height": 50,
              }
            }
          />
          <AGRButton
            onPress={[Function]}
            text="editar"
          />
        </View>,
        "style": Object {
          "backgroundColor": "#FFF",
          "flex": 1,
          "flexDirection": "column",
          "padding": 50,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 30,
                  "marginBottom": 15,
                }
              }
            />,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "marginBottom": 15,
                }
              }
            >
              CRM/Matricula:
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "marginBottom": 15,
                }
              }
            >
              setor: 
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "marginBottom": 15,
                }
              }
            >
              Cirurgi達o
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 20,
                  "marginBottom": 15,
                }
              }
            >
              Hospital do Gama
            </Text>,
            <View
              style={
                Object {
                  "height": 50,
                }
              }
            />,
            <AGRButton
              onPress={[Function]}
              text="editar"
            />,
          ],
          "style": Object {
            "alignSelf": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": undefined,
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "center",
                "fontSize": 30,
                "marginBottom": 15,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "CRM/Matricula:",
                undefined,
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              },
            },
            "ref": null,
            "rendered": Array [
              "CRM/Matricula:",
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "setor: ",
                undefined,
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              },
            },
            "ref": null,
            "rendered": Array [
              "setor: ",
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Cirurgi達o",
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              },
            },
            "ref": null,
            "rendered": "Cirurgi達o",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Hospital do Gama",
              "ellipsizeMode": "tail",
              "style": Object {
                "alignSelf": "center",
                "fontSize": 20,
                "marginBottom": 15,
              },
            },
            "ref": null,
            "rendered": "Hospital do Gama",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "style": Object {
                "height": 50,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onPress": [Function],
              "text": "editar",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
      "flexDirection": "column",
      "padding": 50,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 20,
        "marginBottom": 15,
      }
    }
  >
    Carregando...
  </Text>
</View>
`;
